application/utils/session_wrapper.py

```diff
@@ -0,0 +1,42 @@
+from functools import wraps
+
+from flask import current_app
+from application.utils.exceptions import DatabaseError, InvalidTemplateException
+from structlog import get_logger
+
+log = get_logger()
+
+
+def with_db_session(f):
```

Comment:
We could look at using scoped_session http://docs.sqlalchemy.org/en/latest/orm/contextual.html?

Reply:
Tried to see if I could get this working quickly. I'm not sure how much it adds?

Reply:
I guess it means that we don't have to explicitly handle transaction as they are handled at the request level